apis/core/v1alpha2/core_types.go

```diff
@@ -384,6 +386,12 @@ type DataOutput struct {
 
 	// FieldPath refers to the value of an object's field.
 	FieldPath string `json:"fieldPath,omitempty"`
+
+	// Matchers specify the requirements to match a value.
+	// Different requirements are AND-ed together.
+	// If no matcher is specified, it is by default to check not empty.
+	// +optional
+	Matchers []DataMatcherRequirement `json:"matchers,omitempty"`
```

Comment:
If we put this into DataOutput, this doesn't mean "Matchers" anymore.
It is "Conditions":

```
conditions:
- fieldPath: status.state
  op: eq
  value: Running
```

Reply:
And the comment needs to be changed:

```Go
// Conditions specify the conditions that should be satisfied before emitting a data output.
Conditions []ConditionRequirement
```

Reply:
Putting `Conditions` in `dataOutputs` parts seems confusing to me. If the condition turns out to be `false`, I might mistakenly think the outer controller will prevent the deployment of the component. 

Besides, if we set two conditions C1 and C2 for component A, and component B and C depends on component A, there will be confusion when B and C can move on.
By keeping the original design, ie C1 and C2 set in component B and C respectively, if C1 == true, component B can go on; if C2 == false, component C will hold on. This situation seems more efficient. But I admit I could not make a real example for this situation.